### PR TITLE
feat(telemetry): report data-schema version and update staleness

### LINKED ISSUE #1887: Telemetry: version health, report schema_version and update staleness, currently only a point-in-time aoe_version
**Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

Follow-up to #1863 (opt-in usage telemetry), parent epic #1762.

## Problem

We cannot see version health across the install base. Two related blind spots:

1. Update outcomes: the updater checks GitHub for a newer release but emits nothing, so we do not know how stale installs are, whether anyone acts on the "update available" notice, or whether `aoe update` actually ran.
2. Schema / migration state: a `.schema_version` counter drives data migrations, but which version each install sits on, and whether migrations ran or failed, is never reported.

Together these answer "are users on patched versions" and "what is our backward-compat support burden," both of which are currently guesswork. The only version signal today is `aoe_version` in `process_start`, which is a point-in-time string, not staleness or migration state.

## What's there today

Version is carried, but only as a bare string on boot:

```rust
// src/telemetry/events.rs (process_start)
pub aoe_version: String,   // current build only; no "latest", no staleness
```

The updater computes everything needed for a staleness signal and then drops it:

```rust
// src/update/mod.rs (check_for_update fetches latest, compares, caches; emits no telemetry)
```

`print_update_notice` shows a banner but fires no event. So `update_available` is known at runtime and never recorded.

Migrations track a schema version that never leaves the machine:

```rust
// src/migrations/mod.rs:28-29
const CURRENT_VERSION: u32 = 11;
const VERSION_FILE: &str = ".schema_version";
```

`run_migrations()` runs pending migrations and logs at info level only; `TelemetryState` (`src/telemetry/state.rs`) holds just `install_id` and the CLI throttle stamp, no schema version.

## Proposal

A (preferred): add two coarse version-health signals to the snapshot, both already computed locally.

- `schema_version: u32` from `migrations::CURRENT_VERSION` (or the on-disk `.schema_version`). One small integer; reveals the migration-state distribution and how many installs are stuck behind.
- An update-staleness signal from the updater's existing check: at minimum `update_available: bool`; better, a coarse staleness bucket (`current` / `one_behind` / `several_behind`) derived from the cached latest-vs-current comparison the updater already does. Avoid sending the raw latest version string if the bucket suffices; decide with the gateway contract.

Bump `SCHEMA_VERSION`. These are install-level, computed at snapshot-build time (schema_version is a const; update state is already cached by `src/update/mod.rs`), so no new collection path.

B (process_start only): attach `schema_version` to `process_start` instead of the snapshot, so even CLI-only installs report it. Reasonable for schema_version specifically since it is surface-independent; the update-staleness signal fits the snapshot better. Could do both: schema_version on `process_start`, staleness in the snapshot.

C (status quo): infer everything from the `aoe_version` string distribution. Rejected; that shows what people run, not whether they are behind or what migration state they hold.

A (optionally combined with B for schema_version placement) is preferred.

## Why now

- "Are users on patched/recent versions" is a security and support question with no current answer beyond the raw version histogram.
- Schema-version distribution directly sizes the backward-compat burden and flags installs stuck on old data formats.
- Both values are already computed (`CURRENT_VERSION` const, updater cache); this is plumbing, not new detection.

## User stories / regression tests

Implementation MUST add tests covering each story below; PR is not done without them.

- As a maintainer, when an opted-in install is on schema version 11, then the telemetry reports `schema_version = 11`.
  - Suggested test layer: Rust unit asserting the snapshot (or process_start) carries `migrations::CURRENT_VERSION`.
- As a maintainer, when the updater has determined a newer release exists, then the snapshot reflects `update_available = true` (or the agreed staleness bucket).
  - Suggested test layer: Rust unit with an injected/cached update-check result.
- As a privacy reviewer, when version-health signals are sent, then only a small integer and a coarse bucket/bool leave the client, never an arbitrary version string beyond the existing `aoe_version`.
  - Suggested test layer: Rust unit asserting the field types and that no raw "latest" string is emitted if a bucket is chosen.
- As an opted-out user, when an update check runs or a migration executes, then nothing is recorded or sent.
  - Suggested test layer: Rust integration on the default-off path.

## Open questions

- `schema_version` on `process_start` (covers CLI installs) or in the snapshot (TUI/serve only)? It is surface-independent, so `process_start` gives wider coverage.
- Staleness as a bool (`update_available`) or a coarse bucket (`current`/`one_behind`/`several_behind`)? The bucket is more useful and still leaks no exact version.
- Should a migration failure emit a distinct signal (a `migration_failed` flag), or is the schema_version distribution enough to spot stuck installs?

## Out of scope

- The raw "latest available version" string, unless the staleness bucket proves insufficient.
- Update performance/timing or download metrics.
- Crash/panic stability signals, tracked separately.
- Per-migration timing or names; only the resulting schema_version (and optionally a failure flag).

## Cross-references

- #1762 — parent telemetry epic.
- #1863 — PR that introduced `process_start` and the version field.
- #1870 — time-aggregation model any new periodic signal should respect.

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

Adds two privacy-safe version-health signals to the anonymous, opt-in telemetry so the maintainers can answer "are installs on patched/recent versions" and "how big is the backward-compat support burden" without leaking any version string beyond the existing `aoe_version`.

Both events (`process_start` and `usage_snapshot`) now carry three new fields:

- `data_schema_version` (u32): the data-schema version this build targets (`migrations::CURRENT_VERSION`). A failed migration aborts boot, so any running install has converged to it; sending it gives the schema-version distribution without the backend maintaining a version-to-schema map.
- `update_status` (enum): coarse semver-distance bucket from the local update cache, one of `unknown` / `current` / `patch_behind` / `minor_behind` / `major_behind`. `unknown` means no update check has been cached yet and is never reported as `current`.
- `update_releases_behind` (enum): how many cached releases are newer than the running build, one of `unknown` / `current` / `one_behind` / `several_behind`. Counted from the cached release list, so a fallback cache that only knows the latest release reports the conservative `one_behind`.

Both staleness fields read the existing local update-check cache (`update_cache.json`); they never trigger a network call and never include the latest version number, only the coarse gap. Cross-referencing the two reveals cache shape too: `major_behind` + `one_behind` is a thin/fallback cache, `minor_behind` + `several_behind` is a genuinely lagging install. Computation happens only after the opt-in gate, so an opted-out or `DO_NOT_TRACK` install builds and sends nothing.

The fields land on both events on purpose: each event stays self-contained (robust to a dropped `process_start`), and putting them on `process_start` covers CLI-only installs too. `SCHEMA_VERSION` is bumped 1 -> 2.

The choice of the const `migrations::CURRENT_VERSION` over the on-disk `.schema_version`, and a coarse bucket over a bare bool, came out of a `/debate` design pass: under fail-hard migration boot the on-disk value always equals the const for any emitting install, and a bare bool conflates "current" with "never checked".

**Telemetry gateway:** no change needed in `aoe-telemetry`. Its envelope is `extra="allow"` and the sanitizer forwards any number, boolean, or short identifier-string property untouched, so a `u32` and snake_case bucket strings flow through with no collector/ingest edit. (The gateway's `docs/SCHEMA.md` conventional-properties list could be refreshed for hygiene, but that is a separate repo and not required for this to work.)

**PostHog dashboard:** the "aoe usage (v2)" dashboard (1662307) gets new version-health tiles, each tile description naming this PR. They populate once telemetry v2 ships.

Closes #1887

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

- [ ] Enable telemetry (`aoe telemetry enable`), point `AOE_TELEMETRY_ENDPOINT` at a local sink, run `aoe` and confirm `process_start` carries `data_schema_version`, `update_status`, `update_releases_behind`.
- [ ] With no `update_cache.json` present, confirm both events report `update_status: "unknown"` and `update_releases_behind: "unknown"` (never `current`).
- [ ] Seed `update_cache.json` with a `latest_version` newer than the running build and confirm the buckets reflect the gap (e.g. a several-major jump reads `major_behind` / `several_behind`).
- [ ] Confirm no raw latest-version string appears in the emitted payload, only the bucket strings and the integer.
- [ ] With telemetry disabled (or `DO_NOT_TRACK=1`), confirm nothing is built or sent even with a cache and schema present.

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / cockpit (Playwright user story)**
- [x] N/A: this PR doesn't change a user-facing dashboard flow

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle

This is a telemetry-payload change, covered by Rust unit tests (`src/update/mod.rs` pure classifiers) and integration tests (`tests/telemetry.rs`) for the four user stories: schema version reported, staleness buckets reflect an injected cache, no version string leaks, opted-out emits nothing.

## AI Usage

- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**
Investigation, the design debate, and implementation were drafted by Claude under my direction. I made the design calls (both staleness taxonomies, const vs on-disk schema version), reviewed each commit, and own the result.

- [x] I am an AI Agent filling out this form (check box if true)

## Overview

Adds privacy-preserving version-health telemetry to allow maintainers to measure schema-version distribution and coarse update staleness across installs without leaking raw version strings or making network calls.

## What changed (high level)

- New fields added to both process_start and usage_snapshot telemetry events:
  - data_schema_version (u32)
  - update_status (enum: unknown / current / patch_behind / minor_behind / major_behind)
  - update_releases_behind (enum: unknown / current / one_behind / several_behind)
- New public helpers/enums:
  - migrations::current_schema_version()
  - update::UpdateStatus and update::ReleasesBehind
  - update::cached_version_health(current) to compute both signals from the local update cache
- Telemetry event structs extended and telemetry-build paths populate the new fields after the opt-in gate
- Telemetry schema version bumped (wire format extended)
- Tests added/updated:
  - Unit tests for staleness classification and edge cases
  - Integration tests verifying telemetry emission only when opted-in, correct bucketing from injected caches, and no raw-version leakage
- Documentation updated to describe the new “Version health” fields and privacy constraints

## Benefits

- Privacy-first: emits only small integers and short bucket identifiers; raw latest-version strings are never sent
- No extra network activity: staleness is derived from the existing local update-check cache only
- Opt-in only: nothing emitted for opted-out or DO_NOT_TRACK installs
- Each event is self-contained (both process_start and usage_snapshot) so CLI-only runs are covered
- Provides actionable, bounded signals to measure schema rollout and how far installs lag

## Technical notes (concise)

- New classification helpers (classify_update_status, classify_releases_behind) bucket semver distance and count newer cached releases; they handle empty/unparsable caches and "thin" caches conservatively
- Version comparison logic refactored to reuse a shared semver-parts helper
- assemble_usage_snapshot initializes fallback/empty values; build_* functions overwrite with cached results after reading update_cache.json (no network)
- Tests ensure no raw-version leakage (including checking for dotted fragments) and that telemetry is omitted when not opted-in

### CODE REVIEW

File: tests/telemetry.rs [hunk 1]
```diff
@@ -226,3 +248,106 @@ fn unreachable_endpoint_never_blocks() {
...
+        serde_json::to_string(&event).expect("serialize process_start"),
+    ] {
+        assert!(
+            !serialized.contains(secret_latest),
+            "raw cached latest version leaked into payload: {serialized}"
+        );
+        assert!(
+            !serialized.contains("1234"),
+            "a fragment of the cached version leaked: {serialized}"
+        );
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply tests/telemetry.rs:816]: Good catch. Switched the fragment to the dotted slice `1234.5678`, which cannot appear in a dashless hex install-id UUID or any bucket string, so the check stays deterministic. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/1934/commits/4e5ad167fe755503715be65eec402343722cc6db

**Review by njbrake — APPROVED** [review]
